amqp/src/main/scala/akka/stream/alpakka/amqp/javadsl/AmqpFlow.scala

```diff
@@ -0,0 +1,108 @@
+   * either normally or because of an amqp failure.
+   *
+   * @param settings `bufferSize` and `confirmationTimeout` properties are ignored by this connector
+   */
+  def create(
+      settings: AmqpWriteSettings
+  ): akka.stream.javadsl.Flow[WriteMessage, WriteResult, CompletionStage[Done]] =
+    akka.stream.alpakka.amqp.scaladsl.AmqpFlow(settings).mapMaterializedValue(f => f.toJava).asJava
+
+  /**
+   * Creates an `AmqpFlow` that accepts `WriteMessage` elements and emits `WriteResult`.
+   *
+   * This variant of `AmqpFlow` asynchronously waits for message confirmations. Maximum number of messages
+   * simultaneously waiting for confirmation before signaling backpressure is configured with a
+   * `bufferSize` parameter. Emitted results preserve the order of messages pulled from upstream - due to that
+   * restriction this flow is expected to be slightly less effective than it's unordered counterpart.
+   *
+   * In case of upstream failure/finish this stage attempts to process all buffered messages (waiting for
+   * confirmation) before propagating failure/finish downstream.
+   *
+   * This stage materializes to a `CompletionStage` of `Done`, which can be used to know when the Flow completes,
+   * either normally or because of an amqp failure.
+   *
+   * NOTE: This connector uses RabbitMQ's extension to AMQP protocol
+   * ([[https://www.rabbitmq.com/confirms.html#publisher-confirms Publisher Confirms]]), therefore it is not
+   * supposed to be used with another AMQP brokers.
+   */
+  def createWithConfirm(
+      settings: AmqpWriteSettings
+  ): akka.stream.javadsl.Flow[WriteMessage, WriteResult, CompletionStage[Done]] =
+    akka.stream.alpakka.amqp.scaladsl.AmqpFlow
+      .withConfirm(settings = settings)
+      .mapMaterializedValue(_.toJava)
+      .asJava
+
+  /**
+   * Creates an `AmqpFlow` that accepts `WriteMessage` elements and emits `WriteResult`.
+   *
+   * This variant of `AmqpFlow` asynchronously waits for message confirmations. Maximum number of messages
+   * simultaneously waiting for confirmation before signaling backpressure is configured with a
+   * `bufferSize` parameter. Results are emitted downstream as soon as confirmation is received, meaning that
+   * there is no ordering guarantee of any sort.
+   *
+   * In case of upstream failure/finish this stage attempts to process all buffered messages (waiting for
+   * confirmation) before propagating failure/finish downstream.
+   *
+   * This stage materializes to a `CompletionStage` of `Done`, which can be used to know when the Flow completes,
+   * either normally or because of an amqp failure.
+   *
+   * NOTE: This connector uses RabbitMQ's extension to AMQP protocol
+   * ([[https://www.rabbitmq.com/confirms.html#publisher-confirms Publisher Confirms]]), therefore it is not
+   * supposed to be used with another AMQP brokers.
+   */
+  def createWithUnorderedConfirm(
```

Comment:
I'd prefer `createWithConfirmUnordered` as the `unordered` is more a postfix marker to the method.

Reply:
Right, and it seems more in line with naming of `map` operators in akka-streams.